graveler/combined_iterator.go

```diff
@@ -0,0 +1,103 @@
+package graveler
+
+import "bytes"
+
+// CombinedIterator iterates over two listing iterators,
+// in case of duplication (in values or in errors) returns value in iterA
+type CombinedIterator struct {
```

Comment:
All iterators wrappers need to be a private implementation that adheres to the same iterator interface.
In this case - combiedListingIterator is the struct type.
Having a func NewCombinedListingIterator that returns ListingIterator